CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,140 @@
+
+## Abstract
+> A short (~200 word) description of the technical issue being addressed.
+
+## Motivation
+> The motivation is critical for CIPs that want to change the Celo protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the CIP solves. CIP submissions without sufficient motivation may be rejected outright.
+
+The existing implementation of Mento is suitable for facilitating low volume trades, e.g. up to tens of thousands at a time, without incurring slippage greater than a few percent. OTC trading satisfies the needs of medium volume trades, eg ~100k for cUSD and likely much less for stable tokens with lower total supply (like cEUR), but is unable to satisfy high volume CELO <-> stable token trades on the order of millions.
+
+Mento is able to facilitate smaller `CELO <-> stable token` trades with minimal slippage for smaller trade sizes. However, significant slippage of over 2% starts to occur on both Mento and on centralized exchanges at trades sizes of ~$50k+. OTC trading satisfies the needs of medium volume trades, eg ~100k, but similarly takes a hit of around 2-3%.
+
+Users looking to make larger trades are faced with 3 options:
+1. Exchange via Mento slowly over a longer period of time
+2. Arrange one or many OTC trades
+3. Create a large limit order in hopes it gets filled over time
+
+Increasing Mento bucket sizes permanently to allow larger trades via Mento will reduce slippage for (1), but it puts a more significant portion of the Reserve at risk.
+
+Building a process to exchange large (~$1 million+) amounts of CELO for stable tokens will enable large entities to make significant block purchases of stable tokens at scale. While minting of the stable token is expected to be the main use case, reversibility (ie trading `stable token -> CELO`) is desirable to instill higher confidence when trading large quantities.
+
+While Granda Mento will help facilitate `CELO <-> stable token` trades, this is ultimately meant as a mechanism to help purchasers with USD who are seeking to own cUSD. Purchasers would come to an agreement with a broker (eg the Celo Foundation) who owns an existing amount of CELO. The broker would be the one to make the `CELO -> cUSD` trade on-chain.
+
+The proposed implementation is being considered in the short-term for large scale trades until Mento is reworked to support more efficient large-scale minting on a sustained basis.
+
+## Specification
+> The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow for an implementation on the Celo platform (go-celo).
+
+At a high level, the design involves:
+1. Anyone can submit a trade proposal.
+   * The size of the trade is required to be sufficiently large.
+   * The assets being sold in the trade are deposited.
+   * The current oracle price for the trade is recorded.
```

Comment:
Should there be more discretion here? e.g. the contract enforces that the proposed price is within x% of the current oracle rate

Reply:
Is this because there is a desire for the exchange proposer to specify the exact price rather than to rely upon the current oracle median as the price?